packages/core/src/external/carequality/ihe-gateway-v2/report/report.ts

```diff
@@ -0,0 +1,301 @@
+import { QueryTypes, Sequelize } from "sequelize";
+import dayjs from "dayjs";
+import {
+  OutboundPatientDiscoveryResp,
+  XCPDGateway,
+  OutboundDocumentQueryResp,
+  XCAGateway,
+  OutboundDocumentRetrievalResp,
+} from "@metriport/ihe-gateway-sdk";
+
+const patientDiscoveryReportRequests = 10;
+const documentQueryReportRequests = 100;
+const documentRetrievalReportRequests = 100;
+
+type Report = {
+  requestId: string;
+  total: number;
+  successPercentage: number;
+  failurePercentage: number;
+  errorOids: {
+    oid: string;
+    error: string;
+  }[];
+};
+
+type SimplifiedReport = {
+  requestId: string;
+  total: number;
+  successPercentage: number;
+  failurePercentage: number;
+};
+
+type ReportSummary = {
+  simplifiedReports: SimplifiedReport[];
+  averageSuccessPercentage: number;
+  averageFailurePercentage: number;
+  errorDetailsMap: { [key: string]: { count: number; details: string } };
+};
+
+async function getRandomRequestIds({
+  sequelize,
+  tableName,
+  since,
+  limit,
+}: {
+  sequelize: Sequelize;
+  tableName: string;
+  since: string;
+  limit: number;
+}): Promise<string[]> {
+  const query = `
+      SELECT request_id
+      FROM ${tableName}
+      WHERE created_at >= '${since}'
+      ORDER BY created_at DESC
+      LIMIT ${limit};
+    `;
+  const results = await sequelize.query(query, { type: QueryTypes.SELECT });
+  // eslint-disable-next-line
+  return results.map((row: any) => row.request_id);
+}
+
+// execute asynch s
+async function getPatientDiscoveryResultsForRequestId({
+  sequelize,
+  requestId,
+}: {
+  sequelize: Sequelize;
+  requestId: string;
+}): Promise<OutboundPatientDiscoveryResp[]> {
+  const query = `
+        SELECT *
+        FROM patient_discovery_result
+        WHERE request_id = '${requestId}';
+    `;
+  const results = await sequelize.query(query, { type: QueryTypes.SELECT });
+  return results.map(result => {
+    //eslint-disable-next-line
+    return (result as any).data as OutboundPatientDiscoveryResp;
+  });
+}
+
+async function getDocumentQueryResultsForRequestId({
+  sequelize,
+  requestId,
+}: {
+  sequelize: Sequelize;
+  requestId: string;
+}): Promise<OutboundDocumentQueryResp[]> {
+  const query = `
+    SELECT *
+    FROM document_query_result
+    WHERE request_id = '${requestId}';
+`;
+  const results = await sequelize.query(query, { type: QueryTypes.SELECT });
+  return results.map(result => {
+    //eslint-disable-next-line
+    return (result as any).data as OutboundDocumentQueryResp;
```

Comment:
This might not work but for sure is error prone. The casing of prop names we have in the app are diff than the ones on the DB.

Hitting the DB directly is a risk that should only take  with a few guardrails - [example](https://github.com/metriport/metriport/blob/4d6ad2d4f5242bfb528215113f1cb64ecb3382cf/packages/api/src/command/medical/document/check-doc-queries.ts#L198): force compilation errors if we change the name of the column on the raw SQL or vice-versa.

Column names inside a string literal are not protected by the compiler - we can change a column or even table name and these queries wouldn't be flagged by the compiler.

In fact, I just saw the same issue w/ the pooling lambdas' code: [it's just casting from the raw SQL to our type](https://github.com/metriport/metriport/blob/4d6ad2d4f5242bfb528215113f1cb64ecb3382cf/packages/core/src/external/carequality/ihe-gateway/poll-outbound-results.ts#L72)... it's only working because we only access the properties inside the JSONB - but this is very error prone and I think we need to fix it (created a ticket for that one: https://github.com/metriport/metriport-internal/issues/1846).

Reply:
Ok so I added zod schemas to validate that the table results have a value data and that that data value corresponds to one of the schemas defined in the ihe-gateway-sdk. 
The actual tables types are defined in api though so I cant get full typing but this is a big imrpovement.

Reply:
My point was not so much about validating the data has the shape we expect it to have - although this is good, but more that we code it in a way that matches that shape.

The mindset is: how can we code this in a way we get a compilation error if we change something on the API package, like the table or column name, if we don't also update it here?

Example: you created the enums `Tables` and `TableColumns`, but if we change the `request_id` column on the API's model to `req_id`, the compiler won't complain and, thus, this can lead to a runtime error.

This is what we prevented with [this workaround](https://github.com/metriport/metriport/blob/4d6ad2d4f5242bfb528215113f1cb64ecb3382cf/packages/api/src/command/medical/document/check-doc-queries.ts#L201). It's not pretty, but it does the job to move a runtime error to a compilation-time error.

I know this is just reporting, but the solution is fairly simple and it might avoid future headaches - as well as it keeps the pattern, in case someone copies from this one to build something more critical.

Specifically for this case, we prob need to create a `requestIdColumnName` exported const on `packages/api/src/external/carequality/models/outbound-patient-discovery-resp.ts` and a `createdAtColumnName` on `BaseModel` (`packages/api/src/models/_default.ts`).

Reply:
But now I see the issue w/ the models being defined on the API package.

The solution you suggested looks good, so we don't have to make larger changes on the platform.